packages/test/test-end-to-end-tests/src/test/gc/gcContainerRuntimeExpiry.spec.ts

```diff
@@ -47,22 +49,46 @@ import { TestDataObject } from "./mockSummarizerClient";
     );
 
     let container1: IContainer;
-
     const createContainer = async (): Promise<IContainer> => provider.createContainer(runtimeFactory);
+    const loadContainer = async (): Promise<IContainer> => provider.loadContainer(runtimeFactory);
+    let clock: SinonFakeTimers;
 
-    beforeEach(async () => {
-        provider = getTestObjectProvider();
+    before(() => {
+        clock = useFakeTimers();
     });
 
-    it("Container should be closed with a ClientSessionExpired error after the gcSessionExpiryTime is up", async () => {
+    afterEach(() => {
+        clock.reset();
+    });
+
+    after(() => {
+        clock.restore();
+    });
+
+    beforeEach(async () => {
+        provider = getTestObjectProvider();
         container1 = await createContainer();
         container1.on("closed", (error) => {
             assert.strictEqual(error?.errorType, "clientSessionExpiredError");
         });
+    });
 
+    it("Container should be closed with a ClientSessionExpired error after the gcSessionExpiryTime is up", async () => {
         await provider.ensureSynchronized();
-        const delay = async (ms: number) => new Promise((res) => setTimeout(res, ms));
-        await delay(100);
-        assert(container1.closed === true, "Container should be closed");
+        clock.tick(timeoutMs - 1);
+        assert(container1.closed === false, "Container1 should not be closed");
+        clock.tick(1);
+        assert(container1.closed, "Container should be closed");
+    });
+
+    it("Containers should have the same expiry time for the same document", async () => {
+        clock.tick(timeoutMs / 2);
```

Comment:
Can you add comments for each step here to make it clear what is happening. It's a little hard to follow why the test is expecting what it is expecting.

Reply:
Also, it would be better to validate the state of containers just 1 tick before the expiry (similar to the first test case). That will give a more precise validation that the timeouts are the same.

You could also load 2 containers and validate that all 3 of them expire at the same time.